src/highdicom/seg/sop.py

```diff
@@ -944,3 +966,1458 @@ def _encode_pixels(self, planes: np.ndarray) -> bytes:
+        # frame number (if applicable), and the dimension index values of the
+        # segmentation frames
+        # This allows for fairly efficient querying by any of the three
+        # variables: seg frame number, source instance/frame number, segment
+        # number
+        # Column for segment number
+        self._lut_seg_col = 0
+        col_data = [segnum_col_data]
+
+        # Columns for other dimension index values
+        self._lut_dim_ind_cols = {
+            i: ptr for ptr, i in
+            enumerate(self._dim_ind_pointers, len(col_data))
+        }
+        col_data += [
+            indices for indices in dim_index_col_data.values()
+        ]
+
+        # Columns related to source frames, if they are usable for indexing
+        if self._single_source_frame_per_seg_frame:
+            self._lut_src_instance_col = len(col_data)
+            self._lut_src_frame_col = len(col_data) + 1
+            col_data += [
+                source_instance_col_data,
+                source_frame_col_data
+            ]
+
+        # Build LUT from columns
+        self._frame_lut = np.array(col_data).T
+
+    @property
+    def segmentation_type(self) -> SegmentationTypeValues:
+        """highdicom.seg.SegmentationTypeValues: Segmentation type."""
+        return SegmentationTypeValues(self.SegmentationType)
+
+    @property
+    def segmentation_fractional_type(
+        self
+    ) -> Optional[SegmentationFractionalTypeValues]:
+        """
+        highdicom.seg.SegmentationFractionalTypeValues:
+            Segmentation fractional type.
+
+        """
+        if not hasattr(self, 'SegmentationFractionalType'):
+            return None
+        return SegmentationFractionalTypeValues(
+            self.SegmentationFractionalType
+        )
+
+    def iter_segments(self):
+        return iter_segments(self)
+
+    @property
+    def number_of_segments(self) -> int:
+        """int: The number of segments in this SEG image."""
+        return len(self.SegmentSequence)
+
+    @property
+    def segment_numbers(self) -> range:
+        """range: The segment numbers present in the SEG image as a range."""
+        return range(1, self.number_of_segments + 1)
+
+    def get_segment_description(
+        self,
+        segment_number: int
+    ) -> SegmentDescription:
+        """Get segment description for a segment.
+
+        Parameters
+        ----------
+        segment_number: int
+            Segment number for the segment, as a 1-based index.
+
+        Returns
+        -------
+        highdicom.seg.SegmentDescription
+            Description of the given segment.
+
+        """
+        if segment_number < 1 or segment_number > self.number_of_segments:
+            raise IndexError(
+                f'{segment_number} is an invalid segment number for this '
+                'dataset.'
+            )
+        return self.SegmentSequence[segment_number - 1]
+
+    def search_for_segments(
```

Comment:
I don't really have a preference between `find` and `get`, but we should be consistent. Since the SR parsing branch uses get, I changed to `get_segment_numbers`. Resolved in 9f466a42080777e8943d4e00f5f2593a1977adaf